docs/configuration.md

```diff
@@ -0,0 +1,297 @@
+  all bootstrap stakeholders, heavyweight delegation certificates,
+  etc. It makes it clearer which genesis is used, but it doesn't
+  describe _how_ this genesis was constructed. Also it requires more
+  steps to create genesis data, than to create a spec. Current design
+  implies that this mechanism should be used only for mainnet where we
+  want to have full understanding of which addresses have which
+  stake. The format is demonstrated below. `hash` is the hash (Blake2b
+  256 bits) of canonically encoded `mainnet-genesis.json` file. It can
+  be computed manually using `scripts/js/genesis-hash.js` script.
+
+```
+    genesis:
+      src:
+        file: mainnet-genesis.json
+        hash: 5f20df933584822601f9e3f8c024eb5eb252fe8cefb24d1317dc3d432e940ebb
+```
+
+* Another way is to provide a specification how to generate genesis
+  data. It allows one to specify how many nodes should have stake,
+  which bootstrap stakeholders should be used, what should be total
+  balance, etc. When node is launched with spec as genesis, it
+  automatically constructs genesis data from it. This way to specify
+  genesis is supposed to be used for everything except
+  mainnet. Example:
+
+```
+    genesis:
+      spec:
+        initializer:
+          testnetInitializer:
+            testBalance:
+              poors:        12
+              richmen:      4
+              richmenShare: 0.99
+              useHDAddresses: True
+              totalBalance: 600000000000000000
+            fakeAvvmBalance:
+              count: 10
+              oneBalance: 100000
+            distribution:
+              testnetRichmenStakeDistr: []
+            seed: 0
+        blockVersionData:
+          scriptVersion:     0
+          slotDuration:      7000
+          maxBlockSize:      2000000
+          maxHeaderSize:     2000000
+          maxTxSize:         4096 # 4 Kb
+          maxProposalSize:   700 # 700 bytes
+          mpcThd:            0.01 # 1% of stake
+          heavyDelThd:       0.005 # 0.5% of stake
+          updateVoteThd:     0.001 # 0.1% of total stake
+          updateProposalThd: 0.1 # 10% of total stake
+          updateImplicit:    10 # slots
+          softforkRule:
+            initThd:        0.9 # 90% of total stake
+            minThd:         0.6 # 60% of total stake
+            thdDecrement:   0.05 # 5% of total stake
+          txFeePolicy:
+            txSizeLinear:
+              a: 155381 # absolute minimal fees per transaction
+              b: 43.946 # additional minimal fees per byte of transaction size
+          unlockStakeEpoch: 18446744073709551615 # last epoch (maxBound @Word64)
+        protocolConstants: &dev_core_genesis_spec_protocolConstants
+          k: 2
+          protocolMagic: 55550001
+          vssMinTTL: 2
+          vssMaxTTL: 6
+        ftsSeed: "c2tvdm9yb2RhIEdndXJkYSBib3JvZGEgcHJvdm9kYSA="
+        heavyDelegation: {}
+        avvmDistr: {}
+```
```

Comment:
This makes me wonder.

We probably want to keep a close watch on the exact list of differences between the auto-genesis and the `mainnet` one.  This list should not contain any frivolous differences, lest we endanger ourselves with self-induced blindness..

Reply:
I am not sure we want to care much about differences in mainnet genesis and testnet genesis.
From what I know, testnet genesis will be significantly different from the mainnet one (for example, all AVVM balances will be divided by 10). I am not aware of the goal to have as similar genesises as possible.

Reply:
Personally I'm not so concerned about the initial accounts and balances being completely similar to mainnet, but any deviation in the protocol constants should be clearly called out and justified.

Reply:
Clarification: when I said _significantly different_, I meant balances (it wasn't clear indeed).
Almost all other constants should be the same (except e. g. protocol magic and maybe 1-2 more constants).

Reply:
> This would entail maintenance of a documented list of these differences, naturally.

I added lists of differences to `Our configurations` for long-term clusters (2 mainnet stagings and internal staging).